Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -1337,6 +1337,14 @@ void RenderObject::outputRenderObject(TextStream& stream, bool mark, int depth)
             } else
                 stream << " \"" << value.utf8().data() << "\"";
         }
+        if (isSkippedContent())
+            stream << " [skipped-content]";
+        if (style().contentVisibility() != ContentVisibility::Visible) {
+            if (style().contentVisibility() == ContentVisibility::Hidden)
+                stream << " [content-visibility=hidden]";
+            else
+                stream << " [content-visibility=auto]";
```

Comment:
dump lines are getting really long now. I'd prefer it as part of the annotation column.

Reply:
I like that idea, new output is:
```
(B)lock/(I)nline/I(N)line-block, (A)bsolute/Fi(X)ed/(R)elative/Stic(K)y, (F)loating, (O)verflow clip, Anon(Y)mous, (G)enerated, has(L)ayer, hasLayer(S)crollableArea, (C)omposited, Content-visibility:(H)idden/(A)uto, (S)kipped content, (+)Dirty style, (+)Dirty layout
B---YGLSC-- --* RenderView at (0,0) size 800x600 renderer (0x162000ab0) layout box (0x0)
B-----LS--- --    HTML RenderBlock at (0,0) size 800x166 renderer (0x162006920) layout box (0x0) node (0x16201a7a0)
B---------- --      BODY RenderBody at (8,8) size 784x150 renderer (0x162006b90) layout box (0x0) node (0x16201b050)
B--O--L--H- --        DIV RenderBlock at (0,0) size 150x150 renderer (0x162006e00) layout box (0x0) node (0x162005240)
B---YG----S --          RenderBlock at (0,0) size 0x0 renderer (0x162007870) layout box (0x0)
I---------S --            #text RenderText renderer (0x162005d60) layout box (0x0) node (0x1620051c0) length->(11) "\n  Text.\n  "
B---------S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x162007600) layout box (0x0) node (0x1620052d0)
B---YG----S --          RenderBlock at (0,0) size 0x0 renderer (0x162005e20) layout box (0x0)
I---------S --            SPAN RenderInline renderer (0x1620060a0) layout box (0x0) node (0x162005460)
I---------S --              #text RenderText renderer (0x1620062c0) layout box (0x0) node (0x1620053e0) length->(12) "inline child"
I---------S --            #text RenderText renderer (0x162006380) layout box (0x0) node (0x162005580) length->(3) "\n  "
BA----L---S --            DIV RenderBlock at (0,0) size 0x0 renderer (0x162005f60) layout box (0x0) node (0x162005600)
BR----L---S --          DIV RenderBlock at (10,10) size 0x0 renderer (0x162006180) layout box (0x0) node (0x162005690)
BX----L---S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x162006440) layout box (0x0) node (0x162005820)
BR----L---S --          DIV RenderBlock at (0,0) size 0x0 renderer (0x1620066b0) layout box (0x0) node (0x1620058b0)
BA----L---S --            DIV RenderBlock at (0,0) size 0x0 renderer (0x162001d80) layout box (0x0) node (0x162005a40)

Reply:
yeah this looks much better

Reply:
> yeah this looks much better

Thanks! I'll add Tim since he suggested doing this in the first place.

Reply:
Seems to contain all the needed info. Looks good to me.